### PR TITLE
feat(arc-531): add translations modules

### PR DESCRIPTION
**Pull request by bartnaessens**
Following https://github.com/viaacode/hetarchief-proxy/pull/112
- admin translations module: `GET admin/translations`, `POST admin/translations`. Prefixed routes with `admin/` to avoid collision with the public module
- admin module re-uses the internal `site-variables` module

### CODE REVIEW

**Review by ian-emsens-shd — APPROVED** [review]

**Comment by bertyhell** [comment]
I get this error:
```
[Nest] 5064  - 03/31/2022, 2:47:51 PM   ERROR [DataService] GraphQl query failed: [{"extensions":{"path":"$","code":"validation-failed"},"message":"expecting a value for non-nullable variable: \"name\""}] +10ms
[Nest] 5064  - 03/31/2022, 2:47:51 PM   ERROR [DataService] Failed to get data from database +11ms
InternalServerErrorException: Internal Server Error
    at DataService.execute (C:\Users\bert\Documents\studiohyperdrive\repos\hetarchief-proxy\src\modules\data\services\data.service.ts:154:11)
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at SiteVariablesService.getSiteVariable (C:\Users\bert\Documents\studiohyperdrive\repos\hetarchief-proxy\src\modules\admin\site-variables\services\site-variables.service.ts:19:7)
    at async Promise.all (index 1)

```

for this route:
image

**Review by bertyhell — CHANGES_REQUESTED** [review]
see remark above

File: src/modules/admin/translations/dto/translations.dto.spec.ts [hunk 1]
```diff
@@ -0,0 +1,44 @@
+import {
+	CreateVisitDto,
+	UpdateVisitDto,
+	UpdateVisitStatusDto,
+	VisitsQueryDto,
+} from '~modules/visits/dto/visits.dto';
+
+describe('VisitsDto', () => {
```
[the remark this answers is not in the document]
Comment (bartnaessens) [reply src/modules/admin/translations/dto/translations.dto.spec.ts:8]: nope 😂 fixed.

File: src/modules/admin/translations/controllers/translations.controller.spec.ts [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+import { TranslationsController } from './translations.controller';
+
+const mockTranslationsService: Partial<Record<keyof TranslationsService, jest.SpyInstance>> = {
+	getTranslations: jest.fn(),
+	updateTranslations: jest.fn(),
+};
+
+const mockTranslationsResponse = {
+	name: 'translations-frontend',
```
Comment (bertyhell) [inline src/modules/admin/translations/controllers/translations.controller.spec.ts:14]:
```suggestion
	name: TranslationKey.FRONTEND_TRANSLATIONS,
```

File: src/modules/admin/translations/services/translations.service.spec.ts [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+
+	it('services should be defined', () => {
+		expect(translationsService).toBeDefined();
+	});
+
+	describe('getTranslations', () => {
+		it('returns translations', async () => {
+			mockSiteVariablesService.getSiteVariable
+				.mockResolvedValueOnce({
+					name: 'TRANSLATIONS_FRONTEND',
```
Comment (bertyhell) [inline src/modules/admin/translations/services/translations.service.spec.ts:39]:
```suggestion
					name: TranslationKey.FRONTEND_TRANSLATIONS,
```

File: src/modules/admin/translations/services/translations.service.spec.ts [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+					value: {
+						key: 'translation',
+					},
+				})
+				.mockResolvedValueOnce({
+					name: 'TRANSLATIONS_BACKEND',
```
Comment (bertyhell) [inline src/modules/admin/translations/services/translations.service.spec.ts:45]:
```suggestion
					name: TranslationKey.BACKEND_TRANSLATIONS,
```

File: src/modules/admin/translations/dto/translations.dto.ts [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+import { IsEnum, IsObject } from 'class-validator';
+
+import { TranslationKey } from '../types';
+
+export class UpdateTranslationsDto {
+	@ApiProperty({
+		required: true,
+		enum: TranslationKey,
+		description:
+			'The translation set to be updated, possible values: TRANSLATIONS_FRONTEND, TRANSLATIONS_BACKEND',
```
Comment (bertyhell) [inline src/modules/admin/translations/dto/translations.dto.ts:11]:
```suggestion
			'The translation set to be updated, possible values: ' + Object.values(TranslationKey).join(', '),
```

File: src/modules/admin/translations/controllers/translations.controller.ts [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+	private logger: Logger = new Logger(TranslationsController.name, { timestamp: true });
+
+	constructor(private translationsService: TranslationsService) {}
+
+	@Get()
+	public async getTranslations(): Promise<Record<string, Record<string, string>>> {
+		return this.translationsService.getTranslations();
+	}
+
+	@Post()
```
Comment (bertyhell) [inline src/modules/admin/translations/controllers/translations.controller.ts:23]:
```suggestion
	@Post()
	@ApiOperation({
		description: 'Set translations for the specified key. Careful: this overwrites all existing values.',
	})
```

**Review by bertyhell — APPROVED** [review]

**Pull request merged**